src/app/dim-api/reducer.ts

```diff
@@ -410,44 +410,60 @@ export const dimApi = (
   }
 };
 
-function migrateSettings(settings: Settings) {
+/**
+ * Migrates deprecated settings to their new equivalent, and erroneous settings values to their correct value.
+ * This updates the settings state and adds their updates to the update queue
+ */
+function migrateSettings(state: DimApiState) {
   // Fix some integer settings being stored as strings
```

Comment:
personal preference just a thought but a
```ts
const changeSetting2 = <V extends keyof Settings>(prop: V, value: Settings[V]) =>
    changeSetting(state, prop, value);
````
wouldn't go amiss at the top of the inside of migrateSettings, just because the code got what feels like unnecessarily longer and less scannable by a smidge

Reply:
don't want to deal with capturing